src/environment/SCRF.h

```diff
@@ -30,93 +30,78 @@
 #include "qmfunctions/Orbital.h"
 
 namespace mrchem {
+class KAIN;
 /** @class SCRF
  *  @brief class that performs the computation of the  ReactionPotential, named Self Consistent Reaction Field.
  */
-class Nuclei;
-class KAIN;
 class SCRF final {
 public:
-    SCRF(Permittivity e,
-         const Nuclei &N,
+    SCRF(Permittivity &e,
+         const Density &rho_nuc,
          std::shared_ptr<mrcpp::PoissonOperator> P,
          std::shared_ptr<mrcpp::DerivativeOperator<3>> D,
-         double orb_prec,
          int kain_hist,
          int max_iter,
-         bool acc_pot,
          bool dyn_thrs,
          std::string density_type);
     ~SCRF();
-    void UpdateExternalDensity(Density new_density) { this->rho_ext = new_density; }
 
     double setConvergenceThreshold(double prec);
 
     mrcpp::ComplexFunction &getCurrentReactionPotential() { return this->Vr_n; }
-    mrcpp::ComplexFunction &getPreviousReactionPotential() { return this->Vr_nm1; }
-    mrcpp::ComplexFunction &getCurrentDifferenceReactionPotential() { return this->dVr_n; }
-
-    mrcpp::ComplexFunction &getCurrentGamma() { return this->gamma_n; }
-    mrcpp::ComplexFunction &getPreviousGamma() { return this->gamma_nm1; }
-    mrcpp::ComplexFunction &getCurrentDifferenceGamma() { return this->dgamma_n; }
 
     Permittivity &getPermittivity() { return this->epsilon; }
 
     void updateMOResidual(double const err_t) { this->mo_residual = err_t; }
 
     friend class ReactionPotential;
+    void computeEnergies(mrchem::OrbitalVector &Phi);
```

Comment:
I suggest to change the implementation of `computeEnergies` to make it return the values directly, either just electronic and nuclear or electronic, nuclear, and total. I don't see much need to save the values in the class, as they are quite easily recomputable.
```suggestion
    auto computeEnergies(mrchem::OrbitalVector &Phi) -> std::tuple<double, double>;
```

Reply:
And this would possibly need to become a public method.